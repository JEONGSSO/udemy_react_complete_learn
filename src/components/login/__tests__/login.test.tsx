import React from 'react';
import { MockedProvider } from '@apollo/client/testing';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';

import { Login } from '../index';

describe('login test', () => {
  it('로그인 성공 테스트', async () => {
    const email: String = 'test@naver.com';
    const password: String = '1234';

    render(<Login />);

    const emailElem = await screen.findByPlaceholderText(
      '이메일을 입력해주세요.'
    );
    const passwordElem = await screen.findByPlaceholderText(
      '비밀번호를 입력해주세요.'
    );
    fireEvent.change(emailElem, { target: { value: email } });
    fireEvent.change(passwordElem, { target: { value: password } });

    const buttonElem = await screen.findByRole('button', {
      name: '로그인',
    });
    fireEvent.click(buttonElem);

    const loginSuccessElem = await screen.findByText('로그인 성공!');
    expect(loginSuccessElem).toBeInTheDocument();
  });
});
