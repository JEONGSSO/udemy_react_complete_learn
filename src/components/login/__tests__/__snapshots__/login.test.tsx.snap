// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login test 로그인 테스트 1`] = `
<div
  role="textt"
>
  <div />
  <div />
</div>
`;
